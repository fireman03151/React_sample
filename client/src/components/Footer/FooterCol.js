import React from 'react';
import PropTypes from 'prop-types';

import Link from '../helpers/Link';
import { dasherize } from '../../../../utils/slugs';

function FooterCol({ title, links }) {
  return (
    <div className={`footer-col ${dasherize(title)}`}>
      {title ? (
        <div className={`col-header `}>{title}</div>
      ) : (
        <div className='col-spacer'></div>
      )}
      {links.map(({ to, text, internal }, i) => (
        <Link external={!internal} key={`link-${i}`} to={to}>
          {text}
        </Link>
      ))}
    </div>
  );
}

const propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      to: PropTypes.string.isRequired,
      text: PropTypes.string,
      internal: PropTypes.bool
    })
  ).isRequired,
  title: PropTypes.string
};

FooterCol.propTypes = propTypes;
FooterCol.displayName = 'FooterCol';
export default FooterCol;
