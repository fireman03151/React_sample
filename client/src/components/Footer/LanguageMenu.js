/* eslint-disable jsx-a11y/no-onchange */
import React from 'react';
import { useTranslation } from 'react-i18next';

const {
  availableLangs,
  i18nextCodes,
  langDisplayNames
} = require('../../../i18n/allLangs');
const { homeLocation } = require('../../../config/env');

const locales = availableLangs.client;

const LanguageMenu = () => {
  const { i18n, t } = useTranslation();
  const i18nLanguage = i18n.language;

  const currentLanguage = Object.keys(i18nextCodes).find(
    key => i18nextCodes[key] === i18nLanguage
  );

  const changeLanguage = e => {
    const path = window.location.pathname;

    if (e.target.value === 'espanol') {
      window.location.replace(`${homeLocation}/espanol${path}`);
    } else {
      window.location.replace(`${homeLocation}${path}`);
    }
  };

  return (
    <div className='language-menu'>
      <label>
        {t('footer.language')}
        <select onChange={e => changeLanguage(e)} value={currentLanguage}>
          {locales.map((lang, i) => {
            return (
              <option key={i} value={lang}>
                {langDisplayNames[lang]}
              </option>
            );
          })}
        </select>
      </label>
    </div>
  );
};

export default LanguageMenu;
