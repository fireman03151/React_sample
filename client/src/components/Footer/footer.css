/* 11. Site Footer
/* ---------------------------------------------------------- */

.site-footer {
  color: var(--tertiary-color);
  background: var(--tertiary-background);
  line-height: 1.6;
  padding: 40px 15px;
  font-size: 16px;
}

.footer-top {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(30.5em, 1fr));
}

@media (max-width: 500px) {
  .footer-top {
    grid-template-columns: repeat(auto-fit, minmax(19em, 1fr));
  }
}

.footer-top,
.footer-bottom {
  width: min(100%, 1300px);
  margin-inline: auto;
}

.site-footer p {
  margin: 0 0 1.45rem;
  line-height: 30px;
  font-size: 16px;
}

.site-footer a {
  color: var(--tertiary-color);
  text-decoration: none;
  padding: 2px;
}

.site-footer a:hover {
  text-decoration: underline;
}

.site-footer .col-header,
.trending-guides .col-header {
  text-align: center;
  font-weight: 700;
  font-size: 16px;
  padding: 0 15px 15px;
  margin-bottom: 0;
}

.trending-guides-articles {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(12em, 1fr));
  gap: 0.5rem;
}

.trending-guides-articles {
  white-space: pre;
}

.footer-row {
  margin: 0;
}

.footer-col {
  padding-inline: 15px;
  font-size: 16px;
}

.footer-col a {
  padding: 5px 0;
}

.footer-desc-col {
  padding-inline: 15px;
  margin-bottom: 30px;
  align-self: center;
}

.footer-desc-col a {
  text-decoration: underline;
  padding: 2px;
}

p.footer-donation {
  font-weight: 700;
  font-size: 18px;
}

p.footer-donation a:hover {
  text-decoration: none;
}

.trending-guides {
  margin: 0 0 3rem;
}

.footer-bottom .our-nonprofit {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.footer-bottom .our-nonprofit a {
  padding: 5px 10px;
}

@media (min-width: 500px) {
  .footer-col {
    font-size: 15;
  }
}

@media (min-width: 800px) {
  .footer-col {
    font-size: 16.5;
  }
  .footer-bottom .our-nonprofit a {
    padding: 5px;
  }
}

@media (min-width: 1200px) {
  .footer-desc-col {
    flex: 1 0 45%;
  }
  .trending-guides {
    flex: 1 0 58%;
  }
  p.footer-donation {
    margin-top: 40px;
  }
  .footer-bottom .our-nonprofit {
    padding-inline: 10px;
    padding-top: 1.25rem;
    justify-content: space-between;
  }
  .footer-bottom .col-header {
    visibility: hidden;
    padding: 0px;
    margin: 0;
    height: 1px;
  }
  .our-nonprofit {
    border-top: 1px;
    border-top-style: solid;
    border-top-color: var(--quaternary-background);
  }
}
