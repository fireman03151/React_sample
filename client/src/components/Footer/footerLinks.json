{
  "footerLinks": [
    {
      "title": "Our Nonprofit",
      "links": [
        { "to": "/news/about/", "text": "About" },
        {
          "to": "https://www.linkedin.com/school/free-code-camp/people/",
          "text": "Alumni Network"
        },
        { "to": "https://github.com/freeCodeCamp/", "text": "Open Source" },
        { "to": "/news/shop/", "text": "Shop" },
        { "to": "/news/support/", "text": "Support" },
        { "to": "/news/sponsors/", "text": "Sponsors" },
        {
          "to": "/news/academic-honesty-policy/",
          "text": "Academic Honesty"
        },
        { "to": "/news/code-of-conduct/", "text": "Code of Conduct" },
        { "to": "/news/privacy-policy/", "text": "Privacy Policy" },
        { "to": "/news/terms-of-service/", "text": "Terms of Service" },
        { "to": "/news/copyright-policy/", "text": "Copyright Policy" }
      ]
    },
    {
      "title": "Trending Guides",
      "links": [
        {
          "to": "/news/2019-web-developer-roadmap/",
          "text": "2019 Web Developer Roadmap"
        },
        { "to": "/news/best-python-tutorial/", "text": "Python Tutorial" },
        {
          "to": "/news/understanding-flexbox-everything-you-need-to-know-b4013d4dc9af/",
          "text": "CSS Flexbox Guide"
        },
        {
          "to": "/news/best-javascript-tutorial/",
          "text": "JavaScript Tutorial"
        },
        { "to": "/news/python-example/", "text": "Python Example" },
        { "to": "/news/best-html-html5-tutorial/", "text": "HTML Tutorial" },
        {
          "to": "/news/linux-command-line-bash-tutorial/",
          "text": "Linux Command Line Guide"
        },
        { "to": "/news/javascript-example/", "text": "JavaScript Example" },

        { "to": "/news/best-git-tutorial/", "text": "Git Tutorial" },
        {
          "to": "/news/best-react-javascript-tutorial/",
          "text": "React Tutorial"
        },
        { "to": "/news/best-java-8-tutorial/", "text": "Java Tutorial" }
      ]
    },
    {
      "title": "",
      "links": [
        { "to": "/news/the-best-linux-tutorials/", "text": "Linux Tutorial" },

        { "to": "/news/best-css-and-css3-tutorial/", "text": "CSS Tutorial" },
        { "to": "/news/the-best-jquery-examples/", "text": "jQuery Example" },
        { "to": "/news/best-sql-database-tutorial/", "text": "SQL Tutorial" },
        { "to": "/news/css-example-css3/", "text": "CSS Example" },
        { "to": "/news/react-examples-reactjs/", "text": "React Example" },
        {
          "to": "/news/best-angular-tutorial-angularjs/",
          "text": "Angular Tutorial"
        },
        {
          "to": "/news/the-best-bootstrap-examples/",
          "text": "Bootstrap Example"
        },
        {
          "to": "/news/the-ultimate-guide-to-ssh-setting-up-ssh-keys/",
          "text": "How to Set Up SSH Keys"
        },
        {
          "to": "/news/best-wordpress-tutorial/",
          "text": "WordPress Tutorial"
        },
        { "to": "/news/the-best-php-examples/", "text": "PHP Example" }
      ]
    }
  ]
}
