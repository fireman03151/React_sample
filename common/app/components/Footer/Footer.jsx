import React from 'react';
import { Col, Row, Grid } from 'react-bootstrap';

import links from './links.json';

export default class extends React.Component {
  static displayName = 'Footer';
  renderLinks(mobile) {
    return links.map(link => {
      return (
        <a
          className={ link.className}
          href={ link.href }
          key={ link.content }
          target={ link.target }>
          { this.renderContent(mobile, link.content) }
        </a>
      );
    });
  }

  renderContent(mobile, content) {
    if (mobile) {
      return (
        <span className='sr-only'>
          content;
        </span>
      );
    }
    return content;
  }

  render() {
    return (
      <Grid className='fcc-footer'>
        <Row>
          <Col
            className='hidden-xs hidden-sm'
            xs={ 12 }>
            { this.renderLinks() }
          </Col>
          <Col
            className='visible-xs visible-sm'
            xs={ 12 }>
            { this.renderLinks(true) }
          </Col>
        </Row>
      </Grid>
    );
  }
}
