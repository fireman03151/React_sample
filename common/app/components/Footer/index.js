export { default as Footer } from './Footer.jsx';
