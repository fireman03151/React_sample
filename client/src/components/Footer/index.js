import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import Link from '../helpers/Link';
import LanguageMenu from './LanguageMenu';
import './footer.css';

const { showLocaleDropdownMenu = false } = require('../../../config/env');

const propTypes = {
  children: PropTypes.any
};

const ColHeader = ({ children, ...other }) => (
  <div className='col-header' {...other}>
    {children}
  </div>
);
ColHeader.propTypes = propTypes;

function Footer() {
  const { t } = useTranslation();

  return (
    <footer className='site-footer'>
      <div className='footer-container'>
        <div className='footer-top'>
          <div className='footer-desc-col'>
            {showLocaleDropdownMenu ? <LanguageMenu /> : null}
            <p>{t('footer.tax-exempt-status')}</p>
            <p>{t('footer.mission-statement')}</p>
            <p>{t('footer.donation-initiatives')}</p>
            <p className='footer-donation'>
              {t('footer.donate-text')}{' '}
              <Link className='inline' to='/donate'>
                {t('footer.donate-link')}
              </Link>
              .
            </p>
          </div>
          <div className='trending-guides'>
            <div className='col-header'>{t('footer.trending-guides')}</div>
            <div className='trending-guides-row'>
              <div className='footer-col footer-col-1'>
                <Link external={false} to={t('trending:article1link')}>
                  {t('trending:article1title')}
                </Link>
                <Link external={false} to={t('trending:article2link')}>
                  {t('trending:article2title')}
                </Link>
                <Link external={false} to={t('trending:article3link')}>
                  {t('trending:article3title')}
                </Link>
                <Link external={false} to={t('trending:article4link')}>
                  {t('trending:article4title')}
                </Link>
                <Link external={false} to={t('trending:article5link')}>
                  {t('trending:article5title')}
                </Link>
                <Link external={false} to={t('trending:article6link')}>
                  {t('trending:article6title')}
                </Link>
                <Link external={false} to={t('trending:article7link')}>
                  {t('trending:article7title')}
                </Link>
                <Link external={false} to={t('trending:article8link')}>
                  {t('trending:article8title')}
                </Link>
                <Link external={false} to={t('trending:article9link')}>
                  {t('trending:article9title')}
                </Link>
                <Link external={false} to={t('trending:article10link')}>
                  {t('trending:article10title')}
                </Link>
              </div>
              <div className='footer-col footer-col-2'>
                <Link external={false} to={t('trending:article11link')}>
                  {t('trending:article11title')}
                </Link>
                <Link external={false} to={t('trending:article12link')}>
                  {t('trending:article12title')}
                </Link>
                <Link external={false} to={t('trending:article13link')}>
                  {t('trending:article13title')}
                </Link>
                <Link external={false} to={t('trending:article14link')}>
                  {t('trending:article14title')}
                </Link>
                <Link external={false} to={t('trending:article15link')}>
                  {t('trending:article15title')}
                </Link>
                <Link external={false} to={t('trending:article16link')}>
                  {t('trending:article16title')}
                </Link>
                <Link external={false} to={t('trending:article17link')}>
                  {t('trending:article17title')}
                </Link>
                <Link external={false} to={t('trending:article18link')}>
                  {t('trending:article18title')}
                </Link>
                <Link external={false} to={t('trending:article19link')}>
                  {t('trending:article19title')}
                </Link>
                <Link external={false} to={t('trending:article20link')}>
                  {t('trending:article20title')}
                </Link>
              </div>
              <div className='footer-col footer-col-3'>
                <div className='footer-left'>
                  <Link external={false} to={t('trending:article21link')}>
                    {t('trending:article21title')}
                  </Link>
                  <Link external={false} to={t('trending:article22link')}>
                    {t('trending:article22title')}
                  </Link>
                  <Link external={false} to={t('trending:article23link')}>
                    {t('trending:article23title')}
                  </Link>
                  <Link external={false} to={t('trending:article24link')}>
                    {t('trending:article24title')}
                  </Link>
                  <Link external={false} to={t('trending:article25link')}>
                    {t('trending:article25title')}
                  </Link>
                </div>

                <div className='footer-right'>
                  <Link external={false} to={t('trending:article26link')}>
                    {t('trending:article26title')}
                  </Link>
                  <Link external={false} to={t('trending:article27link')}>
                    {t('trending:article27title')}
                  </Link>
                  <Link external={false} to={t('trending:article28link')}>
                    {t('trending:article28title')}
                  </Link>
                  <Link external={false} to={t('trending:article29link')}>
                    {t('trending:article29title')}
                  </Link>
                  <Link external={false} to={t('trending:article30link')}>
                    {t('trending:article30title')}
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className='footer-bottom'>
          <div className='col-header'>{t('footer.our-nonprofit')}</div>
          <div className='footer-divder' />
          <div className='our-nonprofit'>
            <Link
              external={false}
              to={'https://www.freecodecamp.org/news/about/'}
            >
              {t('footer.links.about')}
            </Link>
            <Link
              external={false}
              sameTab={false}
              to={'https://www.linkedin.com/school/free-code-camp/people/'}
            >
              {t('footer.links.alumni')}
            </Link>
            <Link external={false} to={'https://github.com/freeCodeCamp/'}>
              {t('footer.links.open-source')}
            </Link>
            <Link
              external={false}
              sameTab={false}
              to={'https://www.freecodecamp.org/shop/'}
            >
              {t('footer.links.shop')}
            </Link>
            <Link
              external={false}
              to={'https://www.freecodecamp.org/news/support/'}
            >
              {t('footer.links.support')}
            </Link>
            <Link
              external={false}
              to={'https://www.freecodecamp.org/news/sponsors/'}
            >
              {t('footer.links.sponsors')}
            </Link>
            <Link
              external={false}
              to={'https://www.freecodecamp.org/news/academic-honesty-policy/'}
            >
              {t('footer.links.honesty')}
            </Link>
            <Link
              external={false}
              to={'https://www.freecodecamp.org/news/code-of-conduct/'}
            >
              {t('footer.links.coc')}
            </Link>
            <Link
              external={false}
              to={'https://www.freecodecamp.org/news/privacy-policy/'}
            >
              {t('footer.links.privacy')}
            </Link>
            <Link
              external={false}
              to={'https://www.freecodecamp.org/news/terms-of-service/'}
            >
              {t('footer.links.tos')}
            </Link>
            <Link
              external={false}
              to={'https://www.freecodecamp.org/news/copyright-policy/'}
            >
              {t('footer.links.copyright')}
            </Link>
          </div>
        </div>
      </div>
    </footer>
  );
}

Footer.displayName = 'Footer';
export default Footer;
