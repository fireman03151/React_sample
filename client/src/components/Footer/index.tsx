import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import Link from '../helpers/link';
import './footer.css';

function Footer(): JSX.Element {
  const { t } = useTranslation();

  return (
    <footer className='site-footer'>
      <div className='footer-top'>
        <div className='footer-desc-col'>
          <p>{t('footer.tax-exempt-status')}</p>
          <p>{t('footer.mission-statement')}</p>
          <p>{t('footer.donation-initiatives')}</p>
          <p className='footer-donation'>
            <Trans i18nKey='footer.donate-text'>
              You can
              <Link className='inline' to='/donate'>
                make a tax-deductible donation here
              </Link>
              .
            </Trans>
          </p>
        </div>
        <div className='trending-guides'>
          <h2 className='col-header'>{t('footer.trending-guides')}</h2>
          <div className='trending-guides-articles'>
            {/* the span elements were add so the link can be inline and its text would be the only active target */}
            <span>
              <Link external={false} to={t('trending:article0link')}>
                {t('trending:article0title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article1link')}>
                {t('trending:article1title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article2link')}>
                {t('trending:article2title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article3link')}>
                {t('trending:article3title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article4link')}>
                {t('trending:article4title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article5link')}>
                {t('trending:article5title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article6link')}>
                {t('trending:article6title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article7link')}>
                {t('trending:article7title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article8link')}>
                {t('trending:article8title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article9link')}>
                {t('trending:article9title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article10link')}>
                {t('trending:article10title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article11link')}>
                {t('trending:article11title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article12link')}>
                {t('trending:article12title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article13link')}>
                {t('trending:article13title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article14link')}>
                {t('trending:article14title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article15link')}>
                {t('trending:article15title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article16link')}>
                {t('trending:article16title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article17link')}>
                {t('trending:article17title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article18link')}>
                {t('trending:article18title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article19link')}>
                {t('trending:article19title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article20link')}>
                {t('trending:article20title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article21link')}>
                {t('trending:article21title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article22link')}>
                {t('trending:article22title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article23link')}>
                {t('trending:article23title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article24link')}>
                {t('trending:article24title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article25link')}>
                {t('trending:article25title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article26link')}>
                {t('trending:article26title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article27link')}>
                {t('trending:article27title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article28link')}>
                {t('trending:article28title')}
              </Link>
            </span>
            <span>
              <Link external={false} to={t('trending:article29link')}>
                {t('trending:article29title')}
              </Link>
            </span>
          </div>
        </div>
      </div>
      <div className='footer-bottom'>
        <h2 className='col-header'>{t('footer.our-nonprofit')}</h2>
        <div className='our-nonprofit'>
          <Link external={false} to={t('links:footer.about-url')}>
            {t('footer.links.about')}
          </Link>
          <Link
            external={false}
            sameTab={false}
            to={'https://www.linkedin.com/school/free-code-camp/people/'}
          >
            {t('footer.links.alumni')}
          </Link>
          <Link external={false} to={'https://github.com/freeCodeCamp/'}>
            {t('footer.links.open-source')}
          </Link>
          <Link
            external={false}
            sameTab={false}
            to={t('links:footer.shop-url')}
          >
            {t('footer.links.shop')}
          </Link>
          <Link external={false} to={t('links:footer.support-url')}>
            {t('footer.links.support')}
          </Link>
          <Link external={false} to={t('links:footer.sponsors-url')}>
            {t('footer.links.sponsors')}
          </Link>
          <Link external={false} to={t('links:footer.honesty-url')}>
            {t('footer.links.honesty')}
          </Link>
          <Link external={false} to={t('links:footer.coc-url')}>
            {t('footer.links.coc')}
          </Link>
          <Link external={false} to={t('links:footer.privacy-url')}>
            {t('footer.links.privacy')}
          </Link>
          <Link external={false} to={t('links:footer.tos-url')}>
            {t('footer.links.tos')}
          </Link>
          <Link external={false} to={t('links:footer.copyright-url')}>
            {t('footer.links.copyright')}
          </Link>
        </div>
      </div>
    </footer>
  );
}

Footer.displayName = 'Footer';
export default Footer;
