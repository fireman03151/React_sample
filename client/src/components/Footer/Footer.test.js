/* global expect */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-dom/extend-expect';

import Footer from './';

describe('<Footer />', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
