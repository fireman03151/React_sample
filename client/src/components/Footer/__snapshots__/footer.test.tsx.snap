// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<footer
  className="site-footer"
>
  <div
    className="footer-top"
  >
    <div
      className="footer-desc-col"
    >
      <p>
        footer.tax-exempt-status
      </p>
      <p>
        footer.mission-statement
      </p>
      <p>
        footer.donation-initiatives
      </p>
      <p
        className="footer-donation"
      >
        You can
        <a
          className="inline"
          href="/donate"
        >
          make a tax-deductible donation here
        </a>
        .
      </p>
    </div>
    <div
      className="trending-guides"
    >
      <h2
        className="col-header"
      >
        footer.trending-guides
      </h2>
      <div
        className="trending-guides-articles"
      >
        <span>
          <a
            href="trending:article0link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article0title
          </a>
        </span>
        <span>
          <a
            href="trending:article1link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article1title
          </a>
        </span>
        <span>
          <a
            href="trending:article2link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article2title
          </a>
        </span>
        <span>
          <a
            href="trending:article3link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article3title
          </a>
        </span>
        <span>
          <a
            href="trending:article4link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article4title
          </a>
        </span>
        <span>
          <a
            href="trending:article5link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article5title
          </a>
        </span>
        <span>
          <a
            href="trending:article6link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article6title
          </a>
        </span>
        <span>
          <a
            href="trending:article7link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article7title
          </a>
        </span>
        <span>
          <a
            href="trending:article8link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article8title
          </a>
        </span>
        <span>
          <a
            href="trending:article9link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article9title
          </a>
        </span>
        <span>
          <a
            href="trending:article10link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article10title
          </a>
        </span>
        <span>
          <a
            href="trending:article11link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article11title
          </a>
        </span>
        <span>
          <a
            href="trending:article12link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article12title
          </a>
        </span>
        <span>
          <a
            href="trending:article13link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article13title
          </a>
        </span>
        <span>
          <a
            href="trending:article14link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article14title
          </a>
        </span>
        <span>
          <a
            href="trending:article15link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article15title
          </a>
        </span>
        <span>
          <a
            href="trending:article16link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article16title
          </a>
        </span>
        <span>
          <a
            href="trending:article17link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article17title
          </a>
        </span>
        <span>
          <a
            href="trending:article18link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article18title
          </a>
        </span>
        <span>
          <a
            href="trending:article19link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article19title
          </a>
        </span>
        <span>
          <a
            href="trending:article20link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article20title
          </a>
        </span>
        <span>
          <a
            href="trending:article21link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article21title
          </a>
        </span>
        <span>
          <a
            href="trending:article22link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article22title
          </a>
        </span>
        <span>
          <a
            href="trending:article23link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article23title
          </a>
        </span>
        <span>
          <a
            href="trending:article24link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article24title
          </a>
        </span>
        <span>
          <a
            href="trending:article25link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article25title
          </a>
        </span>
        <span>
          <a
            href="trending:article26link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article26title
          </a>
        </span>
        <span>
          <a
            href="trending:article27link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article27title
          </a>
        </span>
        <span>
          <a
            href="trending:article28link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article28title
          </a>
        </span>
        <span>
          <a
            href="trending:article29link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article29title
          </a>
        </span>
      </div>
    </div>
  </div>
  <div
    className="footer-bottom"
  >
    <h2
      className="col-header"
    >
      footer.our-nonprofit
    </h2>
    <div
      className="our-nonprofit"
    >
      <a
        href="links:footer.about-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.about
      </a>
      <a
        href="https://www.linkedin.com/school/free-code-camp/people/"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.alumni
      </a>
      <a
        href="https://github.com/freeCodeCamp/"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.open-source
      </a>
      <a
        href="links:footer.shop-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.shop
      </a>
      <a
        href="links:footer.support-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.support
      </a>
      <a
        href="links:footer.sponsors-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.sponsors
      </a>
      <a
        href="links:footer.honesty-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.honesty
      </a>
      <a
        href="links:footer.coc-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.coc
      </a>
      <a
        href="links:footer.privacy-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.privacy
      </a>
      <a
        href="links:footer.tos-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.tos
      </a>
      <a
        href="links:footer.copyright-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.copyright
      </a>
    </div>
  </div>
</footer>
`;
